edb/pgsql/common.py

```diff
@@ -81,7 +81,10 @@ def needs_quoting(string):
                string.replace('_', 'a').isalnum())
     return (
         not isalnum or
-        string.lower() in pg_keywords.by_type[pg_keywords.RESERVED_KEYWORD] or
+        string.lower() in pg_keywords.by_type[
+            pg_keywords.RESERVED_KEYWORD] or
+        string.lower() in pg_keywords.by_type[
+            pg_keywords.TYPE_FUNC_NAME_KEYWORD] or
```

Comment:
There's also `COL_NAME_KEYWORD`.

Reply:
Oh, I tried that and I got this during initialization:
```
2. asyncpg.exceptions.UndefinedObjectError: type smallint does not exist


    query, line 127, in query
        | 
        > CREATE  FUNCTION edgedb._derive_uuid(id uuid, variant "smallint")
          ^
        | RETURNS  uuid
...
```

Reply:
Incidentally, creating a type or a link with name `smallint` doesn't cause any problems (maybe I should add that as another test case).

Reply:
Sigh, this is crazy.  They basically hardcoded a bunch of alternative spellings for types as keywords and even though you can do `"int2"`, you cannot do `"smallint"`.  Makes little sense to me, but fine, not quoting `COL_NAME_KEYWORD` is actually fine, since the only place where they require quoting in SQL are unqualified function names, and all our function calls are fully-qualifed.